### PR TITLE
Open points are fixed

### PR DESCRIPTION
**Pull request by ezgicatalbas**
Open points are fixed

### CODE REVIEW

File: src/chat/panes/chat-pane/controller.js [hunk 1]
```diff
@@ -1,32 +1,32 @@
...
+            this.screenTypes = screenTypes
+        });
         CardService.bind('UptodateScreen', () => {
-            this.uptodateScreen = CardService.uptodateScreen;
+            this.screenType = this.screenTypes.UPTODATE;
         })
         CardService.bind('ArchiveScreen', () => {
-            this.archiveScreen = CardService.archiveScreen;
+            this.screenType = this.screenTypes.ARCHIEVE;
         })
```
Comment (FrostTiger) [inline src/chat/panes/chat-pane/controller.js]: ARCHIEVE must be "ARCHIVE".

File: src/chat/root/template.pug [hunk 2]
```diff
@@ -1,4 +1,3 @@
-menu
-mother-pane
-div.logincontainer(ng-if = "$ctrl.setLoginScreen")
```
Comment (FrostTiger) [inline src/chat/root/template.pug]: Third and fourth lines activate login screen. why did you delete these lines.

File: src/lib/mock-backend.js [hunk 3]
```diff
@@ -1134,6 +1134,15 @@ var owner = {
...
 
+var ScreenTypes = {
+        UPTODATE: 1,
+        ARCHIEVE: 2,
+        RECYCLEBIN: 3,
+        USER: 4,
+        PROFILE: 5,
+        SEARCH: 6
+    }
+
```
Comment (FrostTiger) [inline src/lib/mock-backend.js:1145]: `        this.EventType = {
            SET_ACTIVE_THREAD: 'set active thread',
            UPDATE: 'update',
            SET_ACTIVE_CHAT_BOX: 'set active chat box',
            NEW_MESSAGE: 'new message',
            ADD_CHAT_BOX: 'add chat box',
            REMOVE_CHAT_BOX: 'remove chat box'
        };`
@ezgicatalbas we can define ENUM in service card or wherever you want. Teacher has already share with us like above code. This will be more meaningful instead using array.

you can examine it in week-3-2 in card-service.js

File: src/chat/card-service.js [hunk 4]
```diff
@@ -102,36 +61,27 @@ class CardService extends EventEmitter {
...
-        this.trigger('RecycleBin');
-        this.trigger('ArchiveScreen');
-        this.trigger('User');
     }
 
-    searchResult(content) {
-        var te = this.owner.cards;
+    setScreenType(type) {
+        this.screenType = type;
+        if (type == 1) {
```
Comment (dashersw) [inline src/chat/card-service.js]: How about having only two lines here,

```
this.screenType = type;
this.trigger('navigation');
```
do you actually need multiple events for this? This would also simplify your event handlers; instead of binding to multiple events everywhere, you would just bind to the `navigation` event and do `this.screenType = CardService.screenType`.

File: src/chat/chat-boxes/shopping-card/controller.js [hunk 5]
```diff
@@ -5,6 +5,10 @@ module.exports = class ShoppingCardController {
         ChatService.owner.then(owner => {
             this.owner = owner
         });
+        ChatService.screenTypes.then(screenTypes => {
+            debugger;
```
Comment (dashersw) [inline src/chat/chat-boxes/shopping-card/controller.js]: You should remove this debugger statement.

**Review by dashersw — CHANGES_REQUESTED** [review]

File: src/index.js [hunk 6]
```diff
@@ -8,6 +8,7 @@ angular
         $httpBackend.whenGET('/api/threads').respond(mockBackend.getThreads);
         $httpBackend.whenGET('/api/updates').respond(mockBackend.getUpdates);
         $httpBackend.whenGET('/api/owner').respond(mockBackend.getOwner);
+        $httpBackend.whenGET('/api/screenTypes').respond(mockBackend.getScreenTypes);
```
Comment (dashersw) [inline src/index.js:11]: Why is this an API call? Screen types are static, right? They don't depend on user actions or account. You could save yourself the trouble of making an API call to fetch screen types, as they never vary.

File: src/chat/panes/profile/style.css [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    font-size: larger;
+    display: -webkit-inline-box;
+    margin-left: 20px;
+    margin-right: 20px;
+}
+
+.address{
+    margin-left: 20px;
+    font-size: x-large;
+}
```
Comment (dashersw) [inline src/chat/panes/profile/style.css:45]: you need to re-format this document.

File: src/chat/panes/left-menu/template.pug [hunk 8]
```diff
@@ -1,8 +1,7 @@
...
-    div.button1(ng-click = "$ctrl.setItem('newShoppingCard')") Yeni Alışveriş Listesi
-    div.button1(ng-class = "{'archiveScreen': $ctrl.uptodateScreen}" ng-click = "$ctrl.setItem('Uptodate')") Mevcut Listeler
-    div.button1(ng-class = "{'archiveScreen': $ctrl.archiveScreen}"  ng-click = "$ctrl.setItem('Archive')") Arşiv
-    div.button1(ng-class = "{'archiveScreen': $ctrl.recycleBinScreen}"  ng-click = "$ctrl.setItem('Delete')") Çöp Kutusu
-    div.button1(ng-class = "{'archiveScreen': $ctrl.userScreen}"  ng-click = "$ctrl.setItem('User')") Kişiler
-
+    div.button1(ng-click = "$ctrl.newShoppingCard()") Yeni Alışveriş Listesi
+    div.button1(ng-class = "{'archiveScreen': $ctrl.screenType == $ctrl.screenTypes.UPTODATE}" ng-click = "$ctrl.setItem($ctrl.screenTypes.UPTODATE)") Mevcut Listeler
+    div.button1(ng-class = "{'archiveScreen': $ctrl.screenType == $ctrl.screenTypes.ARCHIEVE}"  ng-click = "$ctrl.setItem($ctrl.screenTypes.ARCHIEVE)") Arşiv
+    div.button1(ng-class = "{'archiveScreen': $ctrl.screenType == $ctrl.screenTypes.RECYCLEBIN}"  ng-click = "$ctrl.setItem($ctrl.screenTypes.RECYCLEBIN)") Recycle Bin
```
Comment (dashersw) [inline src/chat/panes/left-menu/template.pug]: Are you sure about the `(ng-class = "{'archiveScreen': $ctrl.screenType == $ctrl.screenTypes.RECYCLEBIN}"` declaration here? Why is it always the class `archiveScreen` regardless of the screen type? What does this class do?

Comment (FrostTiger) [reply src/chat/panes/left-menu/template.pug]: it will change background color grey to orange when mouse on item. We may change unique class name for all items

Comment (dashersw) [reply src/chat/panes/left-menu/template.pug]: how about using `:hover` class and doing it in CSS?

File: src/index.js (hunk 6 above)
Comment (FrostTiger) [reply src/index.js:11]: Because I think screen types is defined in mock-backend.js. We may define it in card-service or new screen service as a enum.

File: src/chat/panes/left-menu/template.pug (hunk 8 above)
Comment (FrostTiger) [reply src/chat/panes/left-menu/template.pug]: like following codes;
`.archiveScreen{
    background-color: darkorange;
}
.archiveScreen:hover{
    background-color: orange;`
